SmartDeviceLink/SDLProtocol.m

```diff
@@ -87,15 +97,23 @@ - (UInt8)sdl_retrieveSessionIDforServiceType:(SDLServiceType)serviceType {
 #pragma mark - SDLTransportDelegate
 
 - (void)onTransportConnected {
-    for (id<SDLProtocolListener> listener in self.protocolDelegateTable.allObjects) {
+    NSArray<id<SDLProtocolListener>> *listeners;
+    @synchronized(self.protocolDelegateTable) {
```

Comment:
Correct me if I'm wrong, but I think `NSHashTable` isn't thread-safe. That is, if we create a new array using `allObjects` and another thread updates the table at the same time, then the array will be corrupted. The lock is added to prevent such multi-threading issue. I know such issue is very unlikely to happen though.

In Java, we have a convenient class called `ConcurrentHashMap`.

Reply:
`allObjects` will create a copy of the objects. The array will not be corrupted. Unless you have a concrete issue that can be traced back to this, this is simply slowing down the library for no clear benefit.